geo/src/algorithm/haversine_distance_to_line.rs

```diff
@@ -0,0 +1,122 @@
+use num_traits::FromPrimitive;
+
+use geo_types::{CoordFloat, Line, Point};
+
+use crate::{CrossTrackDistance, HaversineBearing, HaversineDistance};
+
+pub trait HaversineDistanceToLine<T, Rhs> {
```

Comment:
My initial impulse would be to hang this implementation on the existing `HaversineDistance` trait instead of introducing a new trait. Is there a reason it's better to introduce a new trait like this?